jetbrains-core/src/software/aws/toolkits/jetbrains/services/sqs/toolwindow/PollMessagePane.kt

```diff
@@ -2,8 +2,95 @@
 // SPDX-License-Identifier: Apache-2.0
 package software.aws.toolkits.jetbrains.services.sqs.toolwindow
 
+import com.intellij.icons.AllIcons
+import com.intellij.openapi.actionSystem.ActionManager
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.DefaultActionGroup
+import com.intellij.openapi.project.DumbAware
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.SimpleToolWindowPanel
+import software.amazon.awssdk.services.sqs.SqsClient
+import software.amazon.awssdk.services.sqs.model.Message
+import software.amazon.awssdk.services.sqs.model.QueueAttributeName
+import software.aws.toolkits.jetbrains.services.sqs.Queue
+import software.aws.toolkits.resources.message
+import javax.swing.JLabel
 import javax.swing.JPanel
 
-class PollMessagePane {
+class PollMessagePane(
+    private val project: Project,
+    private val client: SqsClient,
+    private val queue: Queue
+) {
     lateinit var component: JPanel
+    lateinit var messagesAvailableLabel: JLabel
+    lateinit var tablePanel: SimpleToolWindowPanel
+    val messagesTable = MessagesTable()
+
+    private fun createUIComponents() {
+        tablePanel = SimpleToolWindowPanel(false, true)
+    }
+
+    init {
+        tablePanel.setContent(messagesTable.component)
+
+        requestMessages()
+        addTotal()
+        addToolbar()
+    }
+
+    private fun requestMessages() {
+        try {
+            val polledMessages: List<Message> = client.receiveMessage {
```

Comment:
This will end up making a network call on the UI thread (since it's called in `init`), this needs to be put in the background (preferentially using coroutines)